### PR TITLE
Fix issue #11582

### PR DESCRIPTION
**Pull request by ericadestoya**
In `url()` method, looks for `isset($url['_ssl'])` first, so that `_ssl` changes are only made if `$url['_full']` is not met. Then, looks for the string `:///` and replaces it with `://`. Has been tested with all combinations of `$url['_ssl']`, `$url['_full']`, and whether or not `App.fullBaseUrl` has been set.

### CODE REVIEW

**Comment by dereuromark** [comment]
A small test case for this would be great. That will avoid future regressions here.

**Comment by markstory** [comment]
Refs #11582

File: src/Routing/Router.php [hunk 1]
```diff
@@ -617,6 +617,12 @@ public static function url($url = null, $full = false)
             return $output;
         }
         if (is_array($url)) {
+            if (isset($url['_ssl'])) {
+                if (!isset($url['_full'])) {
```
Comment (markstory) [inline src/Routing/Router.php]: Is it possible for these conditions to be combined?  
```php
if (isset($url['_ssl']) && !isset($url['_full'])) {
  $url['_scheme'] = ($url['_ssl'] === true) ? 'https' : 'http';
}
unset($url['_ssl']);
```

**Comment by dereuromark** [comment]
ping @ericadeefox

**Comment by markstory** [comment]
I added tests and fixed up some behavior that didn't seem right in 82acd2beef98f2b4adcd46a8335a865d86d3a759. Previously using `_ssl=true, _full=true` you would get non-SSL URLs which didn't sound right to me.

**Pull request merged**